Horus simulator rotary control improvements 4085

joaquins opened the pull request:
A couple of proposal fixes:
- Add keys for rotary control (comma, period)
- Fix for wheelEvent function

3djc commented on companion/src/simulation/simulatordialog.cpp:
```diff
@@ -70,7 +70,10 @@ void SimulatorDialog::updateDebugOutput()
 void SimulatorDialog::wheelEvent (QWheelEvent *event)
 {
-  simulator->wheelEvent(event->delta() > 0 ? 1 : -1);
+   if ( event->delta() != 0)
```
cosmetics : 
if (...) {
  ...
}

3djc commented:
comma and period may not be a great choice for some language (one is lowercase, one is uppercase on a french keyboard for example)

schwabe commented:
simu uses x and z (which are also not ideal)

3djc commented:
schwabe makes a good point, we should have consistency between both (knowing no key combination will be perfect for all keyboard layouts)

joaquins commented:
Thanks for the comments guys. I already fixed the cosmetic issues, and changed the kay assignments to keys Z and X as suggested.

Is there any other step needed to get this pull request accepted, and mark the opened issue as fixed?

3djc commented:
I requires one of us to test it and approved it (or not). But we are having a very busy day, it may take some time

joaquins commented:
Oh ok @3djc, thanks for your answer. No pressure at all, just wondering if I needed to do anything else.

bsongis commented on companion/src/simulation/simulatordialog.cpp:
```diff
@@ -70,7 +70,9 @@ void SimulatorDialog::updateDebugOutput()
 {
-  simulator->wheelEvent(event->delta() > 0 ? 1 : -1);
+   if ( event->delta() != 0) {
+       simulator->wheelEvent(event->delta() > 0 ? 1 : -1);
```
wrong indentation

bsongis commented on companion/src/simulation/simulatordialog.cpp:
```diff
@@ -220,6 +222,13 @@ void SimulatorDialog::keyPressEvent (QKeyEvent *event)
     case Qt::Key_PageUp:    
       buttonPressed = event->key();
       break;
+    case Qt::Key_Z:
```
On FR keyboards this is not good, perhaps is it possible you check the keyboard layout?

rcbebo82 commented:
Works nice. This is really good it was a pain with the mouse wheel!

schwabe commented:
If we move them to x and c (which also helps German Qwertz), could you do the same for simu.cpp to have the two programs consistent?

joaquins commented:
Ping?
I already did the update to simu to have the same key mapping. Let me know if you are expecting something else.

projectkk2glider commented:
I will test this now...

projectkk2glider commented:
There are two issues with Horus in Companion:
 * mouse scroll-wheel works in reverse
 * PgUp works, but PgDown does not, Backspace works as PgDown.

@joaquins would you be willing to solve this two issues also as a part of this PR? Otherwise your chages work good.

joaquins commented:
@projectkk2glider I would need to get a "real" mouse and a full keyboard for those... using laptop trackpad so far :(

projectkk2glider commented:
OK, then I will merge this and we will deal with #4160 later

projectkk2glider commented:
Thanks!

joaquins commented:
Thanks @projectkk2glider !
